pkg/controller/compute/v1beta1/cluster.go

```diff
@@ -0,0 +1,287 @@
+	cr, ok := mg.(*v1beta1.GKECluster)
+	if !ok {
+		return resource.ExternalObservation{}, errors.New(errNotCluster)
+	}
+
+	existing, err := e.cluster.Projects.Locations.Clusters.Get(meta.GetExternalName(mg)).Context(ctx).Do()
+	if err != nil {
+		return resource.ExternalObservation{}, errors.Wrap(resource.Ignore(gcp.IsErrorNotFound, err), errGetCluster)
+	}
+
+	cr.Status.AtProvider = gke.GenerateObservation(*existing)
+	currentSpec := cr.Spec.ForProvider.DeepCopy()
+	gke.LateInitializeSpec(&cr.Spec.ForProvider, *existing)
+	if !reflect.DeepEqual(currentSpec, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return resource.ExternalObservation{}, errors.Wrap(err, errManagedUpdateFailed)
+		}
+	}
+
+	// If boostrap node pool is present then we want to delete it.
+	for _, pool := range existing.NodePools {
+		if pool != nil {
+			if pool.Name == gke.BootstrapNodePoolName {
+				boostrapNP := strings.Join([]string{meta.GetExternalName(mg), gke.BootstrapNodePoolName}, "/")
+				_, err = e.cluster.Projects.Locations.Clusters.NodePools.Delete(boostrapNP).Context(ctx).Do()
+				if err != nil {
+					// If we fail to delete bootstrap node pool we will be requeued implicitly.
+					return resource.ExternalObservation{}, errors.Wrap(resource.Ignore(gcp.IsErrorNotFound, err), errDeleteBootstrapNodePool)
+				}
+				break
+			}
+		}
+	}
+
+	switch cr.Status.AtProvider.Status {
+	case v1beta1.ClusterStateRunning:
+		cr.Status.SetConditions(v1alpha1.Available())
+		resource.SetBindable(cr)
+	case v1beta1.ClusterStateProvisioning:
+		cr.Status.SetConditions(v1alpha1.Creating())
+	case v1beta1.ClusterStateUnspecified, v1beta1.ClusterStateDegraded, v1beta1.ClusterStateError, v1beta1.ClusterStateReconciling:
+		cr.Status.SetConditions(v1alpha1.Unavailable())
+	}
+
+	upToDate, _ := gke.IsUpToDate(&cr.Spec.ForProvider, *existing)
+
+	return resource.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  upToDate,
+		ConnectionDetails: connectionDetails(existing),
+	}, nil
+}
+
+func (e *clusterExternal) Create(ctx context.Context, mg resource.Managed) (resource.ExternalCreation, error) {
+	i, ok := mg.(*v1beta1.GKECluster)
+	if !ok {
+		return resource.ExternalCreation{}, errors.New(errNotCluster)
+	}
+
+	// Generate GKE cluster from resource spec.
+	cluster := gke.GenerateCluster(i.Spec.ForProvider)
+
+	// Insert default node pool for bootstrapping cluster.
+	gke.GenerateNodePoolForCreate(cluster)
+
+	create := &container.CreateClusterRequest{
+		Cluster: cluster,
+	}
+
+	parent := fmt.Sprintf(parentFormat, e.projectID, i.Spec.ForProvider.Location)
+
+	if _, err := e.cluster.Projects.Locations.Clusters.Create(parent, create).Context(ctx).Do(); err != nil {
+		return resource.ExternalCreation{}, errors.Wrap(err, errCreateCluster)
+	}
+
+	clusterName := fmt.Sprintf(clusterNameFormat, e.projectID, i.Spec.ForProvider.Location, i.Spec.ForProvider.Name)
+	meta.SetExternalName(i, clusterName)
```

Comment:
Set external name. On next `Observe` we should be able to successfully get the cluster with `meta.GetExternalName()`.